Common subdirectories: spack-src/cmake/compile_tests and spack-src-fj/cmake/compile_tests
Common subdirectories: spack-src/cmake/deps and spack-src-fj/cmake/deps
diff -u spack-src/cmake/kokkos_corner_cases.cmake spack-src-fj/cmake/kokkos_corner_cases.cmake
--- spack-src/cmake/kokkos_corner_cases.cmake	2020-05-06 01:39:39.000000000 +0900
+++ spack-src-fj/cmake/kokkos_corner_cases.cmake	2020-10-19 14:35:19.000000000 +0900
@@ -4,21 +4,6 @@
   # was actually built into Clang. Otherwise the user will get nonsensical
   # errors when they try to build.
 
-  #Try compile is the height of CMake nonsense
-  #I can't just give it compiler and link flags
-  #I have to hackily pretend that compiler flags are compiler definitions
-  #and that linker flags are libraries
-  #also - this is easier to use than CMakeCheckCXXSourceCompiles
-  TRY_COMPILE(CLANG_HAS_OMP
-    ${KOKKOS_TOP_BUILD_DIR}/corner_cases
-    ${KOKKOS_SOURCE_DIR}/cmake/compile_tests/clang_omp.cpp
-    COMPILE_DEFINITIONS -fopenmp=libomp
-    LINK_LIBRARIES -fopenmp=libomp
-  )
-  IF (NOT CLANG_HAS_OMP)
-    UNSET(CLANG_HAS_OMP CACHE) #make sure CMake always re-runs this
-    MESSAGE(FATAL_ERROR "Clang failed OpenMP check. You have requested -DKokkos_ENABLE_OPENMP=ON, but the Clang compiler does not appear to have been built with OpenMP support")
-  ENDIF()
   UNSET(CLANG_HAS_OMP CACHE) #make sure CMake always re-runs this
 ENDIF()
 
diff -u spack-src/cmake/kokkos_enable_devices.cmake spack-src-fj/cmake/kokkos_enable_devices.cmake
--- spack-src/cmake/kokkos_enable_devices.cmake	2020-05-06 01:39:39.000000000 +0900
+++ spack-src-fj/cmake/kokkos_enable_devices.cmake	2020-10-19 14:36:02.000000000 +0900
@@ -32,7 +32,7 @@
 ENDIF()
 KOKKOS_DEVICE_OPTION(OPENMP ${OMP_DEFAULT} HOST "Whether to build OpenMP backend")
 IF(KOKKOS_ENABLE_OPENMP)
-  SET(ClangOpenMPFlag -fopenmp=libomp)
+  SET(ClangOpenMPFlag -Kopenmp)
   IF(KOKKOS_CLANG_IS_CRAY)
     SET(ClangOpenMPFlag -fopenmp)
   ENDIF()
Common subdirectories: spack-src/cmake/Modules and spack-src-fj/cmake/Modules
Common subdirectories: spack-src/cmake/tpls and spack-src-fj/cmake/tpls
